iceoryx_posh/include/iceoryx_posh/internal/popo/building_blocks/chunk_distributor.inl

```diff
@@ -176,14 +176,19 @@ inline uint64_t ChunkDistributor<ChunkDistributorDataType>::deliverToAllStoredQu
             //          and without this intersection we would deliver to dead queues
             typename MemberType_t::LockGuard_t lock(*getMembers());
             typename ChunkDistributorDataType::QueueContainer_t queueIntersection(remainingQueues.size());
-            std::sort(getMembers()->m_queues.begin(), getMembers()->m_queues.end());
-            std::sort(remainingQueues.begin(), remainingQueues.end());
+            auto greaterThan = [](rp::RelativePointer<ChunkQueueData_t>& a,
+                                  rp::RelativePointer<ChunkQueueData_t>& b) -> bool {
+                return reinterpret_cast<uint64_t>(a.get()) > reinterpret_cast<uint64_t>(b.get());
+            };
+            std::sort(getMembers()->m_queues.begin(), getMembers()->m_queues.end(), greaterThan);
+            std::sort(remainingQueues.begin(), remainingQueues.end(), greaterThan);
 
             auto iter = std::set_intersection(getMembers()->m_queues.begin(),
```

Comment:
I double checked, and it appears that `set_intersection` requires sorted ranges despite in theory being able to sort them itself (by the predicate provided). It is important that the ranges are sorted by the same predicate used in the intersection, which is the case here.

For us it does not even matter much what the order is, just that it is a well-defined order (irreflexive, transitive etc.), which it is if sorting.